packages/engine_umbrella/apps/engine/lib/vehicle.ex

```diff
@@ -88,12 +103,21 @@ defmodule Vehicle do
       |> Map.put_new(:capacity, %{volume: 15, weight: 700})
       |> Map.update(:metadata, nil, &Jason.encode!/1)
 
-    struct(Vehicle, vehicle_fields)
-    |> apply_vehicle_to_state()
-    |> (&%VehicleRegistered{vehicle: &1}).()
-    |> ES.add_event()
+    vehicle = struct(Vehicle, vehicle_fields)
 
-    vehicle_fields.id
+    case Vex.valid?(vehicle) do
```

Comment:
a suggestion to using case is using the [_with_ syntax](https://openmymind.net/Elixirs-With-Statement/)
```
with true <- Vex.valid?(vehicle) do
      vehicle
      |> apply_vehicle_to_state()
      |> (&%VehicleRegistered{vehicle: &1}).()
      |> ES.add_event()

      vehicle_fields.id
    else
      _ ->
        IO.inspect(Vex.errors(vehicle), label: "vehicle validation errors")
        Vex.errors(vehicle)
    end

Reply:
And the same in Booking

Reply:
This `with` is nice